src/e2e-test/java/io/cdap/plugin/s3/stepsdesign/S3.java

```diff
@@ -19,11 +36,59 @@ public void openS3SinkProperties() {
     openSinkPluginProperties("S3");
   }
 
-  @Then("Verify data is transferred to target S3 bucket")
-  public void verifyDataIsTransferredToTargetS3Bucket() {
-    TestSetupHooks.verifyDataIsTransferredToTargetS3Bucket();
+  @Then("Verify csv data is transferred to target S3 bucket")
+  public void verifyCSVDataIsTransferredToTargetS3Bucket() {
+    String s3Bucket = TestSetupHooks.s3TargetBucketName;
+    boolean flag = false;
+    List<String> lst = new ArrayList<>();
+    try {
+      for (S3ObjectSummary s3ObjectSummary : S3Client.listObjects(s3Bucket)) {
+        if (s3ObjectSummary.getKey().contains("part")) {
+          flag = true;
+          S3ObjectInputStream s3ObjectInputStream = S3Client.getObjectContent(s3Bucket, s3ObjectSummary.getKey());
+          readInputStream(s3ObjectInputStream, lst);
+        }
+      }
+      if (flag) {
+        Path path = Paths.get(Objects.requireNonNull(S3.class.getResource
+          ("/" + PluginPropertyUtils.pluginProp("outputCSV"))).getPath()).toAbsolutePath();
+        Assert.assertEquals("Output content should match",
+                            new String(Files.readAllBytes(path)), getSortedCSVContent(lst));
+        BeforeActions.scenario.write("Data transferred to s3 bucket " + s3Bucket + " successfully");
+      } else {
+        Assert.fail("Data not transferred to target s3 bucket " + s3Bucket);
+      }
+    } catch (Exception e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        Assert.fail("Target s3 bucket " + s3Bucket + " not created - " + e.getMessage());
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  private String getSortedCSVContent(List<String> lst) {
+    // Since the spark output files aren't guaranteed to be ordered, the output entry with [id, first_name] schema needs
+    // to be sorted by id for comparison purpose.
+    lst.sort((s1, s2) -> {
+      String id1 = s1.split(",")[0], id2 = s2.split(",")[0];
+      return Integer.parseInt(id1) - Integer.parseInt(id2);
+    });
+    StringBuilder sb = new StringBuilder();
+    for (String s : lst) {
+      sb.append(s);
+    }
+    return sb.toString();
   }
-  
+
+  private void readInputStream(InputStream input, List<String> lst) throws IOException {
+    BufferedReader reader = new BufferedReader(new InputStreamReader(input));
```

Comment:
need to close the reader. use try (BufferedReader reader = new BufferedReader(new InputStreamReader(input)) {}